### PR TITLE
Dev: upgrade serial library

### PR DESCRIPTION
**Pull request by jubeira**
This PR modifies the code to be able to use the latest version of [usb-serial-for-android](https://github.com/mik3y/usb-serial-for-android). 
As the USB library changed the way of opening a port and sending data, this PR changes the API of the Base Devices supported by `base_controller`.

This PR also does a small improvement in device initialization by adding a small common layer to all devices, and renames folders to match the package names properly.

@adamantivm ptal.

### CODE REVIEW

File: base_controller_lib/src/main/java/com/ekumen/base_driver/create/CreateBaseDevice.java [hunk 1]
```diff
@@ -147,4 +135,10 @@ public OdometryStatus getOdometryStatus() {
         return null;
     }
 
+    @Override
+    protected void setConnectionParameters(UsbSerialPort port) throws Exception {
+        port.setParameters(9600, UsbSerialPort.DATABITS_8,
+                UsbSerialPort.STOPBITS_1, UsbSerialPort.PARITY_NONE);
+    }
+
```
Comment (adamantivm) [inline base_controller_lib/src/main/java/com/ekumen/base_driver/create/CreateBaseDevice.java]: tiny nit: extra space

File: base_controller_lib/src/main/java/com/ekumen/base_driver/AbstractBaseDevice.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+package com.ekumen.base_driver;
+
+import android.hardware.usb.UsbDeviceConnection;
+import com.hoho.android.usbserial.driver.UsbSerialPort;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import java.io.IOException;
+
+public abstract class AbstractBaseDevice implements BaseDevice {
```
Comment (adamantivm) [inline base_controller_lib/src/main/java/com/ekumen/base_driver/AbstractBaseDevice.java]: Please add class javadoc

**Review by adamantivm — APPROVED** [review]
awesome! Minor nits and LGTM

**Comment by jubeira** [comment]
Nit addressed, doc added.
Usb library name was changed to the one that was [PRd to rosjava_mvn_repo](https://github.com/rosjava/rosjava_mvn_repo/pull/21), which matches the original name given by the author.
@adamantivm ptal if you want!

**Pull request merged**